src/lib/data/bluetooth.ts

```diff
@@ -0,0 +1,166 @@
+import {Sensor} from '../../components/Sensor';
+import {RealtimeDataModel, SensorModel, db} from '../../database';
+import {
+  cadenceMeter,
+  getAllSensors,
+  heartRateMonitor,
+  powerMeter,
+} from '../sensor';
+import {getOrCreateRealtimeRecord} from './realtime';
+import {Subscription} from 'rxjs';
+
+type Sensor = {
+  id: string;
+  name: string;
+  is_primary: boolean;
+  sensorType: string[];
+  type: string;
+  address: string;
+};
+
+function getSensorFromType(sensorType: string) {
+  switch (sensorType) {
+    case 'HeartRate':
+      return heartRateMonitor;
+
+    case 'CyclingPower':
+      return powerMeter;
+
+    case 'CyclingSpeedAndCadence':
+      return cadenceMeter;
+
+    default:
+      console.log('Unknown sensor type');
+      throw new Error('Unknown sensor type');
+  }
+}
+
+// Find first sensor of specific type in DB. Can't query it because table is JSON :/
+function findFirstSensorOfType(
+  sensors: Sensor[],
+  sensorType: string,
+): Sensor | undefined {
+  return sensors.find(sensor => sensor.sensorType.includes(sensorType));
```

Comment:
I think this is probably fine, but if we were worried about perf we could probably do the query in sqlite with a raw query.

Reply:
yeah we are never gonna have an insane number of types or sensors